enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md

```diff
@@ -315,6 +337,69 @@ type ImagePolicyStatus struct {
 
 ### Implementation Details/Notes/Constraints [optional]
 
+#### Populating PolicyType using a Kuberentes Secret
+
+When the Machine Config Operator (MCO) detects a valid (non-nil) reference to a Kubernetes Secret, `ImagePolicySecret` in `PolicyRootOfTrust`, it will fetch the Secret and validate the keys within it according to the specified `PolicyType`. Based on this validation, the `MCO` will populate the corresponding `PolicyType` fields. For example, if the `PolicyType` is set to `PublicKey`, it will extract and validate the `public-key` and `rekor-key` from the Secret, populating the `PublicKey` struct appropriately. Similarly, for other types like `PKI` or `FulcioCAWithRekor`, the MCO will validate and populate fields from the Secret as needed.
```

Comment:
Can the user still specify some values directly if a secret is used for other values?

If so, what happens if both the CR and the secret contain a value for an item?

Reply:
> Can the user still specify some values directly if a secret is used for other values?

When we read the secret data, we can check if it the required key pairs to properly populate the fields of the given image policy. If the secret has other unrelated keys, we can just ignore them. 

>If so, what happens if both the CR and the secret contain a value for an item?

It is upto us to decide. I vote for k8s secret to override the existing values, because then the reconciler in MCO won't have to do any additional verification. If the secret exists that has the values, we should just override the image policy with it. Customer can always delete that k8s secret to make sure the values don't get overridden.